relation_engine/taxa/gtdb/loaders/gtdb_taxa_delta_loader.py

```diff
@@ -1,102 +1,55 @@
-        'This is typically the same collection for all delta loaded data.')
-    parser.add_argument(
-        '--node-collection',
-        required=True,
-        help='the name of the ArangoDB collection into which taxa nodes will be loaded')
-    parser.add_argument(
-        '--edge-collection',
-        required=True,
-        help='the name of the ArangoDB collection into which taxa edges will be loaded')
-    parser.add_argument(
-        '--load-version',
-        required=True,
-        help='the version of this load. This version will be added to a field in the nodes and ' +
-             'edges and will be used as part of the _key field.')
-    parser.add_argument(
-        '--load-timestamp',
-        type=int,
-        required=True,
-        help='the timestamp to be applied to the load, in unix epoch milliseconds. Any nodes ' +
-             'or edges created in this load will start to exist with this time stamp. ' +
-             'NOTE: the user is responsible for ensuring this timestamp is greater than any ' +
-             'other timestamps previously used to load data into the NCBI taxonomy DB.')
-    parser.add_argument(
-        '--release-timestamp',
-        type=int,
-        required=True,
-        help='the timestamp, in unix epoch milliseconds, when the data was released ' +
-        'at the source.')
-
-    return parser.parse_args()
+    parser.add_argument('--config', required=True,
+                        help='the path to the loader configuration file. NOTE: the config '
+                        + 'file will need to be updated for each consecutive load; it is not '
+                        + 'static.')
+    a = parser.parse_args()
+    with open(a.config, 'rb') as c:
+        return DeltaLoaderConfig(c, [_INPUT_FILE])
 
 
 def main():
-    a = parse_args()
-    client = ArangoClient(hosts=a.arango_url)
-    if a.user:
-        if a.pwd_file:
-            with open(a.pwd_file) as pwd_file:
-                pwd = pwd_file.read().strip()
-        else:
-            pwd = getpass.getpass()
-        db = client.db(a.database, a.user, pwd, verify=True)
+    cfg = get_config()
+    client = ArangoClient(hosts=cfg.url)
+    if cfg.user:
+        db = client.db(cfg.database, cfg.user, cfg.password, verify=True)
     else:
-        db = client.db(a.database, verify=True)
+        db = client.db(cfg.database, verify=True)
     attdb = ArangoBatchTimeTravellingDB(
         db,
-        a.load_registry_collection,
-        a.node_collection,
-        default_edge_collection=a.edge_collection)
+        cfg.load_registry_collection,
+        cfg.node_collection,
+        default_edge_collection=cfg.edge_collection)
 
-    with open(a.file) as in1, open(a.file) as in2:
+    input_file = cfg.inputs[_INPUT_FILE]
+    with open(input_file) as in1, open(input_file) as in2:
         nodeprov = GTDBNodeProvider(in1)
         edgeprov = GTDBEdgeProvider(in2)
 
         load_graph_delta(_LOAD_NAMESPACE, nodeprov, edgeprov, attdb,
```

Comment:
It's not entirely clear to me how GTDBNodeProvider/GTDBEdgeProvider and load_graph_delta behave in the face of errors. It may be out of scope, but I'm not seeing anywhere that catches errors thrown by GTDBNodeProvider/GTDBEdgeProvider in load_graph_delta, nor can I easily tell what the effects there would be.

Reply:
This occured to me because we dont seem to do any validation of (or checking for the existence of) the input files themselves before this `with` block

Reply:
Error handling was deliberately left simple / non-existent here since this is meant only to be run by KBase experts. Hence it just chucks any exceptions right into said expert's face